proposals/split_kyverno_controllers.md

```diff
@@ -0,0 +1,154 @@
+# Meta
+[meta]: #meta
+- Name: Split / Decompose Kyverno Controllers
+- Start Date: November 9th, 2022
+- Update data (optional): November 9th, 2022
+- Author(s): @JimBugwadia, @eddycharly
+
+# Table of Contents
+[table-of-contents]: #table-of-contents
+- [Meta](#meta)
+- [Table of Contents](#table-of-contents)
+- [Overview](#overview)
+- [Definitions](#definitions)
+- [Motivation](#motivation)
+- [Proposal](#proposal)
+- [Implementation](#implementation)
+- [Migration (OPTIONAL)](#migration-optional)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+- [Prior Art](#prior-art)
+- [Unresolved Questions](#unresolved-questions)
+- [CRD Changes (OPTIONAL)](#crd-changes-optional)
+
+# Overview
+[overview]: #overview
+
+Kyverno 1.8 runs as a single Deployment. This includes the webhook, reporting, and background controllers for generate and update requests. This design causes heavy backhround processing to impact the webhook performance. This proposal separates the webhook into its own Deployment.
+
+# Definitions
+[definitions]: #definitions
+
+Make a list of the definitions that may be useful for those reviewing. Include phrases and words that Kyverno users or other interested parties may not be familiar with.
+
+# Motivation
+[motivation]: #motivation
+
+Here are some reasons why we are considering splitting a Kyverno installation into multiple components:
+- Eliminate (or greatly reduce) client-side throttling during processing of admission requests in the webhook
+- Allow independent sizing and scaling of components. For example, some components operate active-active while others operate using a leader-election. Hence, separating these makes sense.
+- Enhance security by limiting permissions for each component. This may not be a main driver, but is inteesting to consider.
+
+# Minimal requirements for a release
+
+The main goal for an initial implementation is to separate all background processing from the webhook processing, to eliminate throttling and allow scaling the webhook to 
+
+# Proposal
+
+## Kyverno Components
+
+Here is a list of the current top-level Kyverno components:
+
+| Controller  | Function      |
+| ------------| ------------- |
+| Webhook     | Processes AdmissionReview requests from the API Server |
+| Reports Controller | Processes AdmissionReport, ClusterAdmissionReport , BackgroundScanReport, and ClusterBackgroundScanReport |
+| Update Controller | Processes UpdateRequests for mutate and generate policies |
+| Policy Controller | Processes policy changes |
```

Comment:
We have combined these two into the background controller that takes care of generate and mutateExisting policies.